### ISSUE TITLE
Transform with "power": "on" happens before the bulb turns on

### ISSUE THREAD

**Issue by Djelibeybi** [delfick/photons-core#7]
There is a mismatch in behaviour between a `lan:transform` with `power: off` and `power:on`. 

When you specify `power:off` then entire transform occurs (i.e. from current to target HSBK) before the power off happens. 

However, when you specify `power:on`, the transform (at least for H/S/K) occurs before the bulb turns on, which means the duration only affects the increase in brightness, not the transformation from the last state the bulb was in prior to powering off.

This has quite a jarring effect if the bulb is transitioned to 1500K at night and then you want it to slowly transition back to 3000K in the morning. It turns on at 3000K, then gets brighter.

**Comment by delfick** [comment]
Yeah, this matches how the http api does it.

When the device is already off and you're turning it on, it will first set brightness to 0, turn it on and then increase the brightness.

I can add an option to the transformer to not reset color when it does that

**Comment by delfick** [comment]
relevant code is here https://github.com/delfick/photons-core/blob/master/photons_control/transform.py#L44

**Comment by Djelibeybi** [comment]
I think having that option for Photons _and_ the HTTP API would be super peachy keen. :)

**Comment by delfick** [comment]
I'll mention that to the team, but I doubt we'll add that option to the API

**Comment by Djelibeybi** [comment]
I'm happy with just having it in Photons, but I'm curious to know why you wouldn't add it to the API (as a non-default, must be specified manually option)?

**Comment by delfick** [comment]
I can't really discuss how we determine what features to add and how those are prioritised

**Comment by Djelibeybi** [comment]
I may have used almost exactly the same line on a customer earlier today, so I've been hoist upon my own petard. 😂 All good.

**Comment by delfick** [comment]
:)

**Comment by Djelibeybi** [comment]
I added a `keep_color` option to the `using` function: https://github.com/delfick/photons-core/compare/master...Djelibeybi:transform

I also updated Interactor to be able to set/send both `keep_brightness` and `keep_color`: https://github.com/delfick/photons-interactor/compare/master...Djelibeybi:transform

Now I'm just working out how to add tests for this.

**Comment by Djelibeybi** [comment]
@delfick if you have a second to see why my `can retain previous color when powering on` test is failing, that would be super peachy keen. I can't seem to work out why the Transfomer is setting the bulb to blue before powering on:

```
d073d5000001
GOT:
	(1024, 101, '{}')
	(1024, 119, '{"brightness": 0.0, "cycles": 1.0, "hue": 249.9977111467155, "kelvin": 0, "period": 0.0, "reserved6": "00", "saturation": 1.0, "set_brightness": 1, "set_hue": 1, "set_kelvin": 0, "set_saturation": 1, "skew_ratio": 0.0, "transient": 0, "waveform": "<Waveform.SAW: 0>"}')
	(1024, 117, '{"duration": 10.0, "level": 65535}')
	(1024, 119, '{"brightness": 0.29999237048905164, "cycles": 1.0, "hue": 249.9977111467155, "kelvin": 0, "period": 10.0, "reserved6": "00", "saturation": 1.0, "set_brightness": 1, "set_hue": 1, "set_kelvin": 0, "set_saturation": 1, "skew_ratio": 0.0, "transient": 0, "waveform": "<Waveform.SAW: 0>"}')

EXPECTED:
	(1024, 101, '{}')
	(1024, 117, '{"duration": 10.0, "level": 65535}')
	(1024, 119, '{"brightness": 0.29999237048905164, "cycles": 1.0, "hue": 249.9977111467155, "kelvin": 0, "period": 10.0, "reserved6": "00", "saturation": 1.0, "set_brightness": 1, "set_hue": 1, "set_kelvin": 0, "set_saturation": 1, "skew_ratio": 0.0, "transient": 0, "waveform": "<Waveform.SAW: 0>"}')
```

There's something I'm missing with how the Pipeline is being created but my monkey hammering hasn't yielded any fruit yet.

**Comment by delfick** [comment]
great start!

I've added comments.

Also can we make `keep_color=False` into `transition_color=True` as `keep_color` sounds like the color after the transformation won't be different.

Also, when you make the PR run in the root of photons

```
$ ./format
$ ./lint -i C901
```

Thanks !

**Comment by Djelibeybi** [comment]
Resolved via #8

**Comment by delfick** [comment]
Fixed in 0.24.7 :)
